### ISSUE TITLE
TS.INCRBY - Feature Request: Allow timestamp param

### ISSUE THREAD

**Issue by OskarHandmark** [RedisTimeSeries/RedisTimeSeries#72]
Hi. Just started using RedisTimeSeries. One feature that would be fantastic is to be able to specify `timestamp` for `INCRBY`.

I'm doing some bucketing on data that needs to be bucketed by their inherent timestamp (not using system clock). Let me know if there's another clever way to accomplish the same thing with the same performance.

### Solution inserting a lot of ones
I don't imagine this scales as well as `INCRBY` would. I'll be wanting to the same thing for a lot of different keys, e.g `test:A:B:C`. Doesn't work if I have multiple entries on the same second.

```
127.0.0.1:6379> TS.ADD test 1554384485 1
OK
127.0.0.1:6379> TS.ADD test 1554384490 1
OK
127.0.0.1:6379> TS.ADD test 1554384500 1
OK
127.0.0.1:6379> TS.ADD test 1554384505 1
OK
127.0.0.1:6379>
127.0.0.1:6379> TS.RANGE test - + AGGREGATION SUM 15
1) 1) (integer) 1554384480
   2) "2"
2) 1) (integer) 1554384495
   2) "2"
```

### Solution with `CREATERULE`

```
127.0.0.1:6379> TS.CREATE test
OK
127.0.0.1:6379> TS.CREATE test:sum
OK
127.0.0.1:6379> TS.CREATERULE test test:sum AGGREGATION sum 15
OK
127.0.0.1:6379> TS.ADD test 1554384485 1
OK
127.0.0.1:6379> TS.ADD test 1554384490 1
OK
127.0.0.1:6379> TS.ADD test 1554384500 1
OK
127.0.0.1:6379> TS.ADD test 1554384505 1
OK
127.0.0.1:6379> TS.RANGE test:sum - +
1) 1) (integer) 1554384480
   2) "2"
2) 1) (integer) 1554384495
   2) "2"
127.0.0.1:6379> TS.GET test:sum
1) (integer) 1554384495
2) "2"
```

### Feature request
Bucket entry according to the specified timestamp, instead of system clock

```
127.0.0.1:6379> TS.INCRBY test 1 TIMESTAMP 1554384410 RESET 15
OK
127.0.0.1:6379> TS.INCRBY test 1 TIMESTAMP 1554384411 RESET 15
OK

# Expected output
127.0.0.1:6379> TS.RANGE test - +
1) 1) (integer) 1554384405
   2) "2"
```

**Comment by K-Jo** [comment]
Suggested commands
```sh
TS.INCRBY key value [TIMESTAMP timestamp] [RESET time-bucket] [RETENTION retentionSecs] [LABELS field value..]
TS.DECRBY key value [TIMESTAMP timestamp] [RESET time-bucket] [RETENTION retentionSecs] [LABELS field value..]
```

**Comment by ashtul** [comment]
@OskarHandmark @gkorland 
Until this is implemented, you could probably call TS.GET, then TS.ADD returnValue + Incrby.

**Comment by ashtul** [comment]
Should we allow for “*”?

**Comment by K-Jo** [comment]
@ashtul yes

**Comment by ashtul** [comment]
We should return timestamp instead of 'OK'.
It will also be consistent with ts.add

**Comment by gkorland** [comment]
I think we should return a tuple, new value+timestamp

**Comment by ashtul** [comment]
For both I guess

**Comment by gkorland** [comment]
Yes, @danni-m @K-Jo ? We should also fix all the clients to support this break.

**Comment by gkorland** [comment]
> @ashtul yes

@danni-m  @K-Jo if we're adding '*' support why do we need the TIMESTAMP keyword?

Shouldn't we just for with this? (* is like the current API, isn't it?)
```
TS.INCRBY key timestamp value [RESET time-bucket] [RETENTION retentionSecs] [LABELS field value..]
```

**Comment by K-Jo** [comment]
Let's take this on Monday

**Comment by vgoklani** [comment]
I would like to see this implemented too, thank you!

**Comment by ashtul** [comment]
#160

**Comment by ashtul** [comment]
@vgoklani @OskarHandmark 
This feature is now merged into master #160

**Comment by OskarHandmark** [comment]
I appreciate the effort you put in to make this work, and it will come to use. thank you.
